response_operations_ui/views/reporting_units.py

```diff
@@ -18,27 +23,50 @@
 @reporting_unit_bp.route('/<ru_ref>', methods=['GET'])
 @login_required
 def view_reporting_unit(ru_ref):
-    ru_details = reporting_units_controllers.get_reporting_unit(ru_ref)
-
-    ru_details['surveys'] = sorted(ru_details['surveys'], key=lambda survey: survey['surveyRef'])
-
-    for survey in ru_details['surveys']:
-        survey['collection_exercises'] = sorted(survey['collection_exercises'],
-                                                key=lambda ce: ce['scheduledStartDateTime'],
-                                                reverse=True)
-
-        for collection_exercise in survey['collection_exercises']:
-            collection_exercise['responseStatus'] = map_ce_response_status(collection_exercise['responseStatus'])
-            statuses = case_controller.get_available_case_group_statuses(survey['shortName'],
-                                                                         collection_exercise['exerciseRef'], ru_ref)
-            collection_exercise['statusChangeable'] = len(statuses['available_statuses']) > 0
-            collection_exercise['companyRegion'] = map_region(collection_exercise['companyRegion'])
+    # Make some initial calls to retrieve some data we'll need
+    reporting_unit = party_controller.get_party_by_ru_ref(ru_ref)
+    cases = case_controller.get_cases_by_business_party_id(reporting_unit['id'])
+    case_groups = case_controller.get_case_groups_by_business_party_id(reporting_unit['id'])
+
+    # Get all collection exercises for retrieved case groups
+    collection_exercise_ids = {case_group['collectionExerciseId'] for case_group in case_groups}
+    collection_exercises = [get_collection_exercise_by_id(ce_id) for ce_id in collection_exercise_ids]
+
+    # Add extra collection exercise details and filter by live date
+    now = datetime.now(timezone.utc)
+    live_collection_exercises = [add_collection_exercise_details(ce, reporting_unit, case_groups)
+                                 for ce in collection_exercises
+                                 if parse_date(ce['scheduledStartDateTime']) < now]
+
+    # Get all related surveys for gathered collection exercises
+    survey_ids = {collection_exercise['surveyId'] for collection_exercise in live_collection_exercises}
+    surveys = [get_survey_by_id(survey_id) for survey_id in survey_ids]
+
+    # Get all respondents for the given ru
+    respondents = [party_controller.get_respondent_party_by_party_id(respondent['partyId'])
+                   for respondent in reporting_unit.get('associations')]
+
+    # Link collection exercises and respondents to appropriate surveys
+    linked_surveys = [survey_with_respondents_and_exercises(survey, respondents, live_collection_exercises, ru_ref)
+                      for survey in surveys]
+    sorted_linked_surveys = sorted(linked_surveys, key=lambda survey: survey['surveyRef'])
+
+    # Add latest active iac code to surveys
```

Comment:
Nice comments 👍

Reply:
I needed them to understand what was happening myself 😂 
The process of getting all the data required for this page seems a bit complicated, we're making a huge amount of api calls for a single page view